### PR TITLE
Upgrade to payment methods and API version 2019-03-14

### PR DESCRIPTION
**Pull request by thorsten-stripe**
* Move card payments to payment methods: https://stripe.com/docs/api/payment_methods
* Upgrade to latest API version: https://stripe.com/docs/upgrades#2019-03-14

### CODE REVIEW

**Comment by mikeshaw-stripe** [comment]
Some nits, otw lgtm

File: server/java/src/main/java/app/fulfillment/Fulfillment.java [hunk 1]
```diff
@@ -49,12 +49,10 @@ public static Integer fulfill(Event event) throws StripeException {
 
             case "payment_intent.payment_failed":
                 // Get the PaymentIntent out from the event.
-                pi = (PaymentIntent) event.getDataObjectDeserializer().getObject();
+                pi = (PaymentIntent) event.getDataObjectDeserializer().deserializeUnsafe();
```
[the remark this answers is not in the document]
Comment (thorsten-stripe) [reply server/java/src/main/java/app/fulfillment/Fulfillment.java]: I was running into issues with getting the object. So in order to not have to upgrade the bindings, I settled for this.

**Comment by michelle-stripe** [comment]
looks great!

**Comment by jeffchan-stripe** [comment]
🎉

**Pull request merged**